src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -65,6 +65,21 @@ public function getConfigTreeBuilder()
                     return $v;
                 })
             ->end()
+            ->beforeNormalization()
+                ->ifTrue(function ($v) {
+                    return isset($v['firewalls']);
+                })
+                ->then(function ($v) {
+                    foreach ($v['firewalls'] as $firewallName => $firewallConfig) {
+                        if (isset($firewallConfig['form_login']['login_path']) && !isset($firewallConfig['form_login']['check_path'])) {
+                            $firewallConfig['form_login']['check_path'] = $firewallConfig['form_login']['login_path'];
+                            $v['firewalls'][$firewallName] = $firewallConfig;
```

Comment:
can be `foreach ($v['firewalls'] as $firewallName => &$firewallConfig) {` and then there is no need for this line?